### ISSUE TITLE
_atuin_search:echoti:5: no such terminfo capability: rmkx

### ISSUE THREAD

**Issue by 33b5e5** [atuinsh/atuin#1023]
Whenever I close Atuin, either by accepting a highlighted entry, or by hitting Esc, I see this in the Terminal:

`_atuin_search:echoti:5: no such terminfo capability: rmkx`

I'm on the latest macOS `13.4 (22F66)`, using `zsh 5.9 (x86_64-apple-darwin22.0)`, and the default macOS Terminal app.

Quick steps to reproduce:
- Ctrl+R
- Esc

**Comment by 33b5e5** [comment]
I was able to resolve this with the following patch, as suggested in https://github.com/marlonrichert/zsh-autocomplete/issues/93:

```diff
diff --git a/atuin/src/shell/atuin.zsh b/atuin/src/shell/atuin.zsh
index b0e160ff..57e6f8b4 100644
--- a/atuin/src/shell/atuin.zsh
+++ b/atuin/src/shell/atuin.zsh
@@ -31,12 +31,16 @@ _atuin_search() {
     zle -I
 
     # Switch to cursor mode, then back to application
-    echoti rmkx
+    if (( ${+terminfo[smkx]} )); then
+        echoti rmkx
+    fi
     # swap stderr and stdout, so that the tui stuff works
     # TODO: not this
     # shellcheck disable=SC2048
     output=$(RUST_LOG=error atuin search $* -i -- $BUFFER 3>&1 1>&2 2>&3)
-    echoti smkx
+    if (( ${+terminfo[smkx]} )); then
+        echoti smkx
+    fi
 
     if [[ -n $output ]]; then
         RBUFFER=""
```

I don't really understand the purpose of the echoti mode switch here, so don't know what I'm missing without it. However, it didn't seem to be working anyway due to the lack of rmkx capability on this system.

**Comment by 33b5e5** [comment]
FYI I'm no longer able to reproduce this. There have been a few updates to both Atuin and macOS, so I have no idea what resolved it, and I just kind of noticed it was no longer happening. Closing.

**Comment by ellie** [comment]
It was likely resolved by #1170! Switching terminal backend meant we no longer needed to mess around with mode switching
